jenkins/catapult/jenkins/catapult-client-build-catapult-project-windows.groovy

```diff
@@ -0,0 +1,241 @@
+								--mode lint \
+								--source-path catapult-src \
+								--linter-path catapult-src/linters
+						"""
+					}
+				}
+				stage('build') {
+					steps {
+						sh "${runDockerBuildCommand}"
+					}
+				}
+				stage('push built image') {
+					when {
+						expression { SHOULD_PUBLISH_BUILD_IMAGE.toBoolean() }
+					}
+					steps {
+						script {
+							docker.withRegistry(DOCKER_URL, DOCKER_CREDENTIALS_ID) {
+								docker.image(buildImageFullName).push()
+							}
+						}
+					}
+				}
+			}
+			post {
+				always {
+					recordIssues enabledForFailure: true, tool: pyLint(pattern: 'catapult-data/logs/pylint.log')
+					recordIssues enabledForFailure: true, tool: pep8(pattern: 'catapult-data/logs/pycodestyle.log')
+					recordIssues enabledForFailure: true, tool: gcc(pattern: 'catapult-data/logs/isort.log', name: 'isort', id: 'isort')
+
+					recordIssues enabledForFailure: true,
+						tool: gcc(pattern: 'catapult-data/logs/shellcheck.log', name: 'shellcheck', id: 'shellcheck')
+				}
+			}
+		}
+		stage('test') {
+			when {
+				expression { isTestEnabled() }
+			}
+			stages {
+				stage('pull dependency images') {
+					when {
+						expression { isCustomTestImage() }
+					}
+					steps {
+						script {
+							docker.withRegistry(DOCKER_URL, DOCKER_CREDENTIALS_ID) {
+								docker.image(buildImageFullName).pull()
+							}
+						}
+					}
+				}
+				stage('run tests') {
+					steps {
+						script {
+							if (isCustomTestImage())
+								testImageName = "registry.hub.docker.com/symbolplatform/symbol-server-test:${buildImageLabel}"
+							else
+								testImageName = "symbolplatform/symbol-server-test:${buildImageLabel}"
+
+							sh """
+								python catapult-src/jenkins/catapult/runDockerTests.py \
+									--image ${testImageName} \
+									--compiler-configuration catapult-src/jenkins/catapult/configurations/${COMPILER_CONFIGURATION}.yaml \
+									--mode ${TEST_MODE} \
+									--verbosity ${TEST_VERBOSITY} \
+									--source-path catapult-src
+							"""
+						}
+					}
+				}
+			}
+		}
+	}
+	post {
+		always {
+			junit '**/catapult-data/logs/*.xml'
+
+			dir('catapult-data') {
+				deleteDir()
+			}
+			dir('mongo') {
+				deleteDir()
+			}
+		}
+	}
+}
+
+Boolean isBuildEnabled() {
+	return 'none' != BUILD_CONFIGURATION
+}
+
+Boolean isTestEnabled() {
+	return 'none' != TEST_MODE
+}
+
+Boolean isManualBuild() {
+	return null != MANUAL_GIT_BRANCH && '' != MANUAL_GIT_BRANCH && 'null' != MANUAL_GIT_BRANCH
+}
+
+Boolean isCustomTestImage() {
+	return '' != TEST_IMAGE_LABEL
+}
+
+String getBranchName() {
+	return isManualBuild() ? MANUAL_GIT_BRANCH : env.GIT_BRANCH
+}
+
+String getBuildImageLabel() {
+	friendlyBranchName = getBranchName()
+	if (0 == friendlyBranchName.indexOf('origin/'))
+		friendlyBranchName = friendlyBranchName.substring(7)
+
+	friendlyBranchName = friendlyBranchName.replaceAll('/', '-')
+	return "catapult-server-${friendlyBranchName}-${env.BUILD_NUMBER}"
+}
```

Comment:
you couldn't refactor this with catapult-client-build-catapult-project.groovy? 

seems main difference is agent - which i think can be parameterize - and lack of code coverage, which can be skipped in win builds

Reply:
Yes, I can refactor into one groovy file.